File: src/AdventureWorks.Application/DataEngine/HumanResources/EmployeePayHistory/Commands/Queries/GetEmployeePayHistories/EmployeePayHistoryLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.HumanResources;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;

namespace AdventureWorks.Application.DataEngine.HumanResources.EmployeePayHistory.Queries.GetEmployeePayHistories
{
    public partial class EmployeePayHistoryLookupModel : IHaveCustomMapping
    {
        public int EmployeeID { get; set; }
        public DateTime RateChangeDate { get; set; }
        public decimal Rate { get; set; }
        public byte PayFrequency { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.HumanResources.EmployeePayHistory, EmployeePayHistoryLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<EmployeePayHistoryLookupModel, BaseEmployeePayHistory>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/human-resources/employee-pay-history/employee-pay-history-lookup-model.ts

export interface IEmployeePayHistoryLookupModel {
  employeeID: number;
  rateChangeDate: Date | string;
  rate: number;
  payFrequency: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class EmployeePayHistoryLookupModel implements IEmployeePayHistoryLookupModel {
  employeeID: number;
  rateChangeDate: Date | string;
  rate: number;
  payFrequency: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<EmployeePayHistoryLookupModel>) {
    if (!!init) {
      Object.assign<EmployeePayHistoryLookupModel, Partial<EmployeePayHistoryLookupModel>>(this, init);
    }
  }
}
